src/Trap.js

```diff
@@ -104,15 +104,13 @@ const activateTrap = () => {
     const workingNode = observed || (lastPortaledElement && lastPortaledElement.portaledElement);
 
     // check if lastActiveFocus is still reachable
-    if (focusOnBody() && lastActiveFocus) {
+    if (focusOnBody() && lastActiveFocus && lastActiveFocus !== document.body) {
```

Comment:
focus on the body and the last element is body?
Can happen.

Unfortunately `lastActiveFocus` is used to hold the state of the Trap, and it is not just "last active element within trap", it's just "last active element recorded".
For `autoFocus:false` cases it will be `body`